### ISSUE TITLE
how to merge lora weight when there are more than 1 model.bin

### ISSUE THREAD

**Issue by hxx-who** [dvlab-research/LISA#90]
Hi, great work!
I want to Merge LoRA Weight of LISA-7B by using the following code
```
python merge_lora_weights_and_save_hf_model.py \
  --version="./LLaVA/LLaVA-Lightning-7B-v1-1"\
  --weight="PATH_TO_pytorch_model.bin" \
  --save_path="./LISA7B"
```
but the ```--weight``` only receives the path to one model.bin，but the model of lisa7B has two as ```pytorch_model-00001-of-00002.bin``` and ```pytorch_model-00002-of-00002.bin```，how can I merge them?

**Comment by hxx-who** [comment]
I aim to train on the base of LISA-7B, and it seems like there's no need to merge.
I tried to set the '''--version''' as the model path of LISA-7B and follow the debug instruction in #85, it finally works though #85 also doesn't know why changing that line would be useful.

**Comment by audreyeternal** [comment]
> Hi, great work! I want to Merge LoRA Weight of LISA-7B by using the following code
> 
> ```
> python merge_lora_weights_and_save_hf_model.py \
>   --version="./LLaVA/LLaVA-Lightning-7B-v1-1"\
>   --weight="PATH_TO_pytorch_model.bin" \
>   --save_path="./LISA7B"
> ```
> 
> but the `--weight` only receives the path to one model.bin，but the model of lisa7B has two as `pytorch_model-00001-of-00002.bin` and `pytorch_model-00002-of-00002.bin`，how can I merge them?

Have you solved this issue? I also had the same question. Thanks!

**Comment by lingcco** [comment]
@audreyeternal me too, and I had four pytorch_mode.bin

**Comment by audreyeternal** [comment]
For those who are still bothered with this problem:

It is highly possible that your deepspeed version is higher than 0.15.2, please refer to this issue: #155 

For the temporary workaround, you can manually load all the shard bins into the model like:
```python
index_file = os.path.join(args.weight, "pytorch_model.bin.index.json")

with open(index_file, "r", encoding="utf-8") as f:
    index_dict = json.load(f)

weight_map = index_dict["weight_map"]  # dict

shard_files = set(weight_map.values()) 
full_state_dict = {}

for shard_file in shard_files:
    shard_path = os.path.join(args.weight, shard_file)
    shard_state_dict = torch.load(shard_path, map_location="cpu")
    full_state_dict.update(shard_state_dict)

model.load_state_dict(full_state_dict, strict=True)

```

**Comment by lingcco** [comment]
> For those who are still bothered with this problem:
> 
> It is highly possible that your deepspeed version is higher than 0.15.2, please refer to this issue: [#155](https://github.com/dvlab-research/LISA/issues/155)
> 
> For the temporary workaround, you can manually load all the shard bins into the model like:
> 
> index_file = os.path.join(args.weight, "pytorch_model.bin.index.json")
> 
> with open(index_file, "r", encoding="utf-8") as f:
>     index_dict = json.load(f)
> 
> weight_map = index_dict["weight_map"]  # dict
> 
> shard_files = set(weight_map.values()) 
> full_state_dict = {}
> 
> for shard_file in shard_files:
>     shard_path = os.path.join(args.weight, shard_file)
>     shard_state_dict = torch.load(shard_path, map_location="cpu")
>     full_state_dict.update(shard_state_dict)
> 
> model.load_state_dict(full_state_dict, strict=True)

Thank you!

**Comment by lingcco** [comment]
@audreyeternal Hello, have you ever encountered the problem that the mask output by eval is blank after fine-tuning? I fine-tuned on my custom dataset (in the format of reasonseg), and I don’t know what went wrong? It was normal before training, but after training I found that the output of the model did not have the **[seg]** token. I suspect this is the reason, but I don’t know how to fix it.



**the format of my dataset:**
{
  "text": [
    "Please provide a detailed analysis of artifacts in this photo, considering physical artifacts (e.g., optical display issues, violations of physical laws, and spatial/perspective errors), structural artifacts (e.g., deformed objects, asymmetry, or distorted text), and distortion artifacts (e.g., color/texture distortion, noise/blur, artistic style errors, and material misrepresentation). Output with interleaved segmentation masks for the corresponding parts of the answer."
  ],
  "is_sentence": true,
  "shapes": [
    {
      "label": "target",
      "labels": [
        "target"
      ],
      "shape_type": "polygon",
      "image_name": "0a1e95ec-d762-4a20-b886-1dceed8e505f.png",
      "points": [
        [
          236.25706940874036,
          159.25964010282777
        ],
        [
          224.41131105398458,
          150.04627249357327
        ],
        [
          229.67609254498714,
          137.54241645244215
        ],
        [
          275.74293059125966,
          134.25192802056554
        ],
        [
          298.7763496143959,
          140.17480719794344
        ],
        [
          293.5115681233933,
          161.89203084832906
        ]
      ],
      "group_id": null,
      "group_ids": [
        null
      ],
      "flags": {}
    }
  ]
}

**Comment by audreyeternal** [comment]
@lingcco Hi, I think this is not related to this issue. Maybe you can raise a new issue for that.
